pkg/render/apiserver.go

```diff
@@ -1089,6 +1086,13 @@ func (c *apiServerComponent) apiServerContainer() corev1.Container {
 			PeriodSeconds: 60,
 		},
 	}
+	// In case of OpenShift, apiserver needs privileged access to write audit logs to host path volume.
+	// Audit logs are owned by root on hosts so we need to be root user and group. Audit logs are supported only in Enterprise version.
+	if c.cfg.Installation.Variant == operatorv1.TigeraSecureEnterprise {
+		apiServer.SecurityContext = securitycontext.NewRootContext(c.cfg.Openshift)
+	} else {
+		apiServer.SecurityContext = securitycontext.NewNonRootContext()
```

Comment:
Need `/tmp` folder permission to be 0777 for Calico v3.27 [apiserver Dockerfile](https://github.com/projectcalico/calico/blob/release-v3.27/apiserver/docker-image/Dockerfile.amd64#L11). The `/tmp/ready` flag will write to `/tmp`. I believe this will work with the main Calico branch.

Reply:
Hi! I made some fix: I have created an `emptyDir` volume for the `/tmp` so it works now:
```
➜ kubectl get pods -n calico-apiserver
NAME                                READY   STATUS    RESTARTS   AGE
calico-apiserver-58fb6b4bd7-66crp   1/1     Running   0          3m28s
calico-apiserver-58fb6b4bd7-w8ls4   1/1     Running   0          3m28s

➜ kubectl get tigerastatus apiserver
NAME        AVAILABLE   PROGRESSING   DEGRADED   SINCE
apiserver   True        False         False      2m58s
```

Reply:
Not sure this is the best, but it can work with the current releases - and hopefully with the future ones. What do you think? Thanks!

Reply:
Maybe we can keep the securityContext change only for main branch and add the emptyDir part when picking to an operator release branch. I believe we don't need to mount the emptyDir for main Calico apiserver due to the reasons I mentioned earlier. An alternative options is to fix the `/tmp` dir permission in Calico v3.27 branch so we don't need to mount the emptyDir.

Reply:
Created https://github.com/projectcalico/calico/pull/8570 for Calico OSS v3.27. We can remove the `/tmp` emptyDir mount from operator once this change gets in.

Reply:
@mihivagyok `calico/apiserver:v3.27.2-19-g7cc1fd061b7f` should have the `/tmp` folder permission fix. Can you give it a try?

Reply:
Hi @hjiawei ! Thank you very much! Sure, I will give a try and let you know!

Reply:
It works!
```
Version:      v3.27.2-19-g7cc1fd061b7f
Build date:   2024-02-29T19:21:31+0000
Git tag ref:  v3.27.2-19-g7cc1fd061
Git commit:   7cc1fd061
...
I0301 16:51:37.450279       1 tlsconfig.go:240] "Starting DynamicServingCertificateController"
I0301 16:51:37.452995       1 run_server.go:80] apiserver is ready.
```

and stays healthy! Thank you very much!

Then I remove my last commit. Or what do you suggest?
Does semaphore get the calico OSS v.3.27 fix automatically?

Thanks!

Reply:
Thank you for testing this @mihivagyok! I think you can remove the last emptyDir commit. calico/apiserver change will be in Calico v3.27.3. It doesn't flow back to current release versions automatically.

Once you have this merged into the operator main branch. You also need to pick it back to operator release v1.33 and v1.32 release branches (may need to confirm with operator maintainers).

Reply:
Thank you!